litex/soc/integration/common.py

```diff
@@ -76,20 +80,290 @@ def get_mem_data(filename_or_regions, endianness="big", mem_size=None):
 def SoCConstant(value):
     return value
 
-class SoCMemRegion:
-    def __init__(self, origin, length, type):
-        assert type in ["cached", "io", "cached+linker", "io+linker"]
-        self.origin = origin
-        self.length = length
-        self.type   = type
 
-    def __str__(self):
-        return "<SoCMemRegion 0x{:x} 0x{:x} {}>".format(
-            self.origin, self.length, self.type)
+class SoCRegion:
```

Comment:
I would find the `SoCRegion` class improvements really useful in LiteX-BuildEnv even if the changes to `litex/soc/integration/soc_core.py` aren't used.